docs/src/main/asciidoc/config-yaml.adoc

```diff
@@ -207,3 +207,58 @@ quarkus:
 
 YAML `null` keys are not included in the assembly of the configuration property name, allowing them to be used
 in any level for disambiguating configuration keys.
+
+== Current Limitation
+The smallrye-config library can only retrieve a single value at a time using a string representation of the path, and it does not provide a built-in way to retrieve complex nested structures in configuration files.  To work around this limitation, sources need to provide a single string representation of each value, effectively flattening the structure to a key-value pair of type `String -> String` . This means that nested structures in JSON or YAML files need to be represented in a flat format.
+
+Although Quarkus primarily uses `.properties` file extension for configuration, the snakeyaml library, which is used for parsing YAML in Quarkus, can also parse JSON structures. This means you can use YAML files with JSON content inside.
+
+To take advantage of this, you can use the application.yaml file (with the .yaml extension) and include JSON structures inside it. Snakeyaml will read both YAML and JSON formats within this file. This allows you to structure your configuration in a more hierarchical manner using JSON inside a YAML file.
+
+Certainly, here's a step-by-step guide on how to use complex configuration structures with Quarkus:
+
+1. Define Your Configuration Interface
+
+[source,java]
+----
+@ConfigMapping(prefix = "server")
+public interface ServiceConfig {
+
+  List<Environment> environments();
+
+  interface Environment {
+    String name();
+    String services();
+  }
+}
+----
+
+2. Install the YAML Extension
```

Comment:
Installation is already covered at the beginning of the guide.

Reply:
this point is removed it